builtin/providers/aws/resource_aws_config_configuration_recorder.go

```diff
@@ -0,0 +1,145 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/configservice"
+)
+
+func resourceAwsConfigConfigurationRecorder() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsConfigConfigurationRecorderPut,
+		Read:   resourceAwsConfigConfigurationRecorderRead,
+		Update: resourceAwsConfigConfigurationRecorderPut,
+		Delete: resourceAwsConfigConfigurationRecorderDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "default",
+				ValidateFunc: validateMaxLength(256),
+			},
+			"role_arn": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateArn,
+			},
+			"recording_group": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"all_supported": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true,
+						},
+						"include_global_resource_types": &schema.Schema{
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"resource_types": &schema.Schema{
+							Type:     schema.TypeSet,
+							Set:      schema.HashString,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
```

Comment:
Do you think we could add validation to this, like ensuring value is one of `configservice.ResourceTypeAwsEc2CustomerGateway`, `configservice.ResourceTypeAwsEc2Eip`, etc?

This way it can be done at validation level, and not when applying.